pkgs/applications/emulators/dosbox-staging/default.nix

```diff
@@ -110,10 +110,14 @@ stdenv.mkDerivation rec {
       existing DOSBox codebase while leveraging modern development tools and
       practices.
     '';
-    license = licenses.gpl2Plus;
-    maintainers = with maintainers; [ joshuafern AndersonTorres ];
-    platforms = platforms.unix;
+    changelog = "https://github.com/dosbox-staging/dosbox-staging/releases/tag/v${self.version}";
+    license = lib.licenses.gpl2Plus;
+    maintainers = [
+      lib.maintainers.joshuafern
+      lib.maintainers.AndersonTorres
+    ];
```

Comment:
stil I think this is worse than just using `with`

Reply:
It is mostly equivalent.

Reply:
> It is mostly equivalent.

This is a non-response. You basically just said No, marked the review as resolved, and merged the pull request 2 hours later. You are not allowing any communication to happen and are disregarding both @aaronArinder's and my reviews.

I am well aware that they are mostly syntactically equivalent. As I said in https://github.com/NixOS/nixpkgs/pull/215860#discussion_r1103713674, right now I don't think there is a better alternative to just using `with maintainers;` here. To me, the repeating uses of `lib.maintainers` makes it less readable, harder to extend or edit, and also takes more space for no apparent benefit. You might have better reasoning behind this change, but "mostly equivalent" is not a valid reason to make the change.

Reply:
The previous change used a not so known builtin functionality of Nixpkgs. It is reasonable that it looks unreadable.

On the other hand, "unreadable" is not applicable to the second example, because it is merely being more explicit. It can be cumbersome, but not too much, given that even the most basic text editors have copy-paste functionality good enough to repeat a couple of words.

Also, `with` should be treated as legacy, and arguments of readability are a bit less convincing when this is put into context.

Reply:
By convention meta.maintainers is mostly one line and expanding the with here just increases code duplication. I don't think this change was necessary and not really justified.

Reply:
> On the other hand, "unreadable" is not applicable to the second example

I never said anything was unreadable, and most things I've said about `builtins.attrValues` still applies. This is still more cumbersome and less readable to me.

> Even the most basic text editors have copy-paste functionality good enough to repeat a couple of words

The same argument can be used to discredit `inherit`, or any syntax sugar in any programming language. Requiring copy-pasting might have a small impact, but it still makes an impact and will make the file just so slightly harder to edit.

> `with` should be treated as legacy

Source needed. Also, this information was not provided when you self-merged this pull request. 

> Arguments of readability are a bit less convincing when this is put into context

As seen in some discussions in https://github.com/NixOS/rfcs/pull/110, using `with` over a large scope of code is seen by an antipattern to a lot of people, but this is barely an issue at such a small scope. To me, the benefits of using `with` here far outweighs the drawbacks or "legacy" as you said, that this syntax might have.

Reply:
>  The same argument can be used to discredit inherit, or any syntax sugar in any programming language.

The point is precisely that `with` is not syntatic sugar. Indeed, if `with` was syntatic sugar, this discussion would not even start.

Further, the use of `inherit` is one of the suggestions of the ["with is an anti-pattern"](https://nix.dev/anti-patterns/language#with-attrset-expression) reference.

Reply:
> The point is precisely that with is not syntatic sugar

I did not say `with` is syntactic sugar. I said `inherit` is, and what you said as an argument against `with` can also be used against `inherit` and other syntactic sugars.

> Further, the use of inherit is one of the suggestions of the ["with is an anti-pattern"](https://nix.dev/anti-patterns/language#with-attrset-expression) reference.

Just to be clear, I think `inherit` is good, and that's my argument against you using `copy-paste` to discredit `with`.

Reply:
> I did not say `with` is syntactic sugar. I said `inherit` is, and what you said as an argument against `with` can also be used against `inherit` and other syntactic sugars.

The point is that `with` and `inherit` are not analogous, therefore your reductio fails to convince.

A true syntatic sugar - as proposed by https://github.com/NixOS/rfcs/pull/110 - is way better than rely on text editor functionalities. I would never in my perfect capacities argue against syntatic sugar for "text editor shenanigans".

The same can't be said about `with`, because albeit `with` is used as syntatic sugar in practice, it truly is not.

> Just to be clear, I think `inherit` is good, and that's my argument against you using `copy-paste` to discredit `with`.

Just to be clear, I think `with` is worse than "text editor shenanigans", and in turn "text editor shenanigans" are worse than syntatic sugar.

Reply:
> The point is that `with` and `inherit` are not analogous

I never said they are. You were making an argument against `with` that could be easily applicable to `inherit` which is why I made the comparison.

> A true syntatic sugar - as proposed by https://github.com/NixOS/rfcs/pull/110 - is way better than rely on text editor functionalities.

This is what I have been saying.

> The same can't be said about `with`, because albeit `with` is used as syntatic sugar in practice, it truly is not.

Again, I never said `with` is a syntactic sugar, and I have been saying that https://github.com/NixOS/rfcs/pull/110 is a better fit for this job.

---

Instead of trying to debate against things I never said, please address the issues I have pointed out:

- Explain and provided sources on why `with` at a small scope like this should be treated as legacy
- Repeating `lib.maintainers` here objectively makes things harder to edit, and I personally find it less readable. Why do the benefits outweigh these drawbacks, or why would you make the change if you think the benefits don't outweigh the drawbacks?

Reply:
> Explain and provided sources

[The source was already provided](https://nix.dev/anti-patterns/language#with-attrset-expression). However it will not convince you.

> on why with at a small scope like this should be treated as legacy

My argument was not about "allow it at small scopes"; it is precisely about allowing it _nowhere_.

> Why do the benefits outweigh these drawbacks

They don't. 

After all, this anti-pattern is more readable than any of its alternatives. 

And if the readability will be pushed as more important than getting rid of anti-patterns, the answer is immediate.